### PR TITLE
Dịch "seq2seq-attention_vn" - Phần 3

### LINKED ISSUE #1741: Dịch "seq2seq-attention_vn" - Phần 3
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_attention-mechanisms/seq2seq-attention_vn.md#L161-L231

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by trGiang99**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1741

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [ ] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 1]
```diff
@@ -164,15 +164,17 @@ out.shape, len(state), state[0].shape, len(state[1]), state[1][0].shape
 ## Training
 -->
 
-## *dịch tiêu đề phía trên*
+## Luyện tập
```
Comment (davidnvq) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:167]:
```suggestion
## Huấn luyện
```

Comment (davidnvq) [reply chapter_attention-mechanisms/seq2seq-attention_vn.md:167]: Từ này trong glossary, em lưu ý dịch những phần về sau nha :3

Comment (davidnvq) [reply chapter_attention-mechanisms/seq2seq-attention_vn.md:167]: Maybe chỗ này nhầm, phía dưới a thấy e dịch đúng ;)

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 2]
```diff
@@ -164,15 +164,17 @@ out.shape, len(state), state[0].shape, len(state[1]), state[1][0].shape
...
 
 <!--
 Similar to :numref:`sec_seq2seq_training`, we try a toy model by applying the same training hyperparameters and the same training loss.
 As we can see from the result, since the sequences in the training dataset are relative short, the additional attention layer does not lead to a significant improvement.
 Due to the computational overhead of both the encoder's and the decoder's attention layers, this model is much slower than the seq2seq model without attention.
 -->
 
-*dịch đoạn phía trên*
+Giống như ở phần :numref:`sec_seq2seq_training`, ta hãy thử một mô hình nhỏ bằng cách sử dụng cùng những siêu tham số huấn luyện và mất mát trong huấn luyện (*training loss*).
```
Comment (davidnvq) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:175]:
```suggestion
Chúng ta hãy xây dựng một mô hình đơn giản sử dụng cùng các siêu tham số huấn luyện và hàm mất mát trên tập huấn luyện như :numref:`sec_seq2seq_training`.
```

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 3]
```diff
@@ -164,15 +164,17 @@ out.shape, len(state), state[0].shape, len(state[1]), state[1][0].shape
...
+Từ kết quả ta có thể thấy, do các chuỗi trong tập huẩn luyện khá ngắn, việc thêm tầng tập trung vào mô hình không tạo ra được cải thiện đáng kể.
```
Comment (davidnvq) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:176]:
```suggestion
Từ kết quả, ta thấy tầng tập trung được thêm vào mô hình không tạo ra cải thiện đáng kể nào do các chuỗi trong tập huấn luyện khá ngắn.
```

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 4]
```diff
@@ -164,15 +164,17 @@ out.shape, len(state), state[0].shape, len(state[1]), state[1][0].shape
...
+Do chi phí tính toán tốn thêm từ các tầng tập trung trong bộ mã hoá và giải mã, mô hình này hoạt động chậm hơn nhiều so với seq2seq không có tầng tập trung.
```
Comment (davidnvq) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:177]:
```suggestion
Do chi phí tính toán tốn thêm từ các tầng tập trung trong bộ mã hoá và bộ giải mã, mô hình này hoạt động chậm hơn nhiều so với mô hình seq2seq không có tầng tập trung.
```

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 5]
```diff
@@ -193,7 +195,7 @@ d2l.train_s2s_ch9(model, train_iter, lr, num_epochs, ctx)
 Last, we predict several sample examples.
 -->
 
-*dịch đoạn phía trên*
+Cuối cùng, ta thử dự đoán một vài ví dụ.
```
Comment (davidnvq) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:198]:
```suggestion
Cuối cùng, ta thử dự đoán một vài mẫu dưới đây.
```

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 6]
```diff
@@ -213,7 +215,8 @@ for sentence in ['Go .', 'Wow !', "I'm OK .", 'I won !']:
 * The decoder of the seq2seq with attention model passes three items from the encoder: the encoder outputs of all timesteps, the hidden state of the encoder's final timestep, and the encoder valid length.
 -->
 
-*dịch đoạn phía trên*
+* Mô hình seq2seq áp dụng cơ chế tập trung thêm một tầng tập trung vào mô hình seq2seq ban đầu.
+* Bộ giải mã của mô hình seq2seq với cơ chế tập trung truyền vào ba tham số từ bộ mã hoá: đầu ra của bộ mã hoá tại tất cả các bước thời gian, trạng thái ẩn của bộ mã hoá tại bước thời gian cuối cùng, chiều dài chuẩn của bộ mã hoá.
```
Comment (davidnvq) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:219]:
```suggestion
* Bộ giải mã của mô hình seq2seq với cơ chế tập trung được truyền vào ba đầu ra từ bộ mã hoá: đầu ra của bộ mã hoá tại tất cả các bước thời gian, trạng thái ẩn của bộ mã hoá tại bước thời gian cuối cùng,  độ dài hợp lệ của bộ mã hoá.
```

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 7]
```diff
@@ -226,7 +229,8 @@ for sentence in ['Go .', 'Wow !', "I'm OK .", 'I won !']:
 2. Can you think of any use cases where `Seq2SeqAttentionDecoder` will outperform `Seq2seqDecoder`?
 -->
 
-*dịch đoạn phía trên*
+1. So sánh `Seq2SeqAttentionDecoder` và `Seq2seqDecoder` bằng cách sử dụng cùng tham số và kiểm tra các mất mát.
+2. Bạn hãy thử suy nghĩ liệu có trường hợp nào mà `Seq2SeqAttentionDecoder` vượt trội hơn so với `Seq2seqDecoder`  ?
```
Comment (davidnvq) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:233]:
```suggestion
2. Bạn hãy thử suy nghĩ liệu có trường hợp nào mà `Seq2SeqAttentionDecoder` vượt trội hơn so với `Seq2seqDecoder`?
```

File: chapter_attention-mechanisms/seq2seq-attention_vn.md (hunk 1 above)
Comment (trGiang99) [reply chapter_attention-mechanisms/seq2seq-attention_vn.md:167]: à đoạn này lúc em dịch em lại nghĩ đây là phần "Bài tập luyện tập" :')

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 8]
```diff
@@ -257,4 +261,4 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 3 -->
-*
+* Đỗ Trường Giang
```
Comment (trGiang99) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:264]:
```suggestion
* Đỗ Trường Giang
* Nguyễn Văn Quang
```

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 9]
```diff
@@ -226,7 +229,8 @@ for sentence in ['Go .', 'Wow !', "I'm OK .", 'I won !']:
 2. Can you think of any use cases where `Seq2SeqAttentionDecoder` will outperform `Seq2seqDecoder`?
 -->
 
-*dịch đoạn phía trên*
+1. So sánh `Seq2SeqAttentionDecoder` và `Seq2seqDecoder` bằng cách sử dụng cùng tham số và kiểm tra các mất mát.
+2. Bạn hãy thử suy nghĩ liệu có trường hợp nào mà `Seq2SeqAttentionDecoder` vượt trội hơn so với `Seq2seqDecoder`?
```
Comment (cuongvng) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:233]:
```suggestion
2. Bạn hãy thử suy nghĩ liệu có trường hợp nào mà `Seq2SeqAttentionDecoder` vượt trội hơn `Seq2seqDecoder`?
```

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 10]
```diff
@@ -164,15 +164,17 @@ out.shape, len(state), state[0].shape, len(state[1]), state[1][0].shape
...
+## Huấn luyện
 
 <!--
 Similar to :numref:`sec_seq2seq_training`, we try a toy model by applying the same training hyperparameters and the same training loss.
 As we can see from the result, since the sequences in the training dataset are relative short, the additional attention layer does not lead to a significant improvement.
 Due to the computational overhead of both the encoder's and the decoder's attention layers, this model is much slower than the seq2seq model without attention.
 -->
 
-*dịch đoạn phía trên*
+Chúng ta hãy xây dựng một mô hình đơn giản sử dụng cùng các siêu tham số huấn luyện và hàm mất mát trên tập huấn luyện như :numref:`sec_seq2seq_training`.
```
Comment (lkhphuc) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:175]:
```suggestion
Chúng ta hãy xây dựng một mô hình đơn giản sử dụng cùng các siêu tham số và hàm mất mát để huấn luyện như :numref:`sec_seq2seq_training`.
```

File: chapter_attention-mechanisms/seq2seq-attention_vn.md [hunk 11]
```diff
@@ -213,7 +215,8 @@ for sentence in ['Go .', 'Wow !', "I'm OK .", 'I won !']:
 * The decoder of the seq2seq with attention model passes three items from the encoder: the encoder outputs of all timesteps, the hidden state of the encoder's final timestep, and the encoder valid length.
 -->
 
-*dịch đoạn phía trên*
+* Mô hình seq2seq áp dụng cơ chế tập trung thêm một tầng tập trung vào mô hình seq2seq ban đầu.
+* Bộ giải mã của mô hình seq2seq với cơ chế tập trung được truyền vào ba đầu ra từ bộ mã hoá: đầu ra của bộ mã hoá tại tất cả các bước thời gian, trạng thái ẩn của bộ mã hoá tại bước thời gian cuối cùng,  độ dài hợp lệ của bộ mã hoá.
```
Comment (lkhphuc) [inline chapter_attention-mechanisms/seq2seq-attention_vn.md:219]:
```suggestion
* Bộ giải mã của mô hình seq2seq với cơ chế tập trung được truyền vào ba đầu ra từ bộ mã hoá: đầu ra của bộ mã hoá tại tất cả các bước thời gian, trạng thái ẩn của bộ mã hoá tại bước thời gian cuối cùng, độ dài hợp lệ của bộ mã hoá.
```
